pkg/runner/cluster_k8s.go

```diff
@@ -692,6 +699,27 @@ func (c *ClusterK8sRunner) createTestplanPod(ctx context.Context, podName string
 			},
 			RestartPolicy: v1.RestartPolicyNever,
 			InitContainers: []v1.Container{
+				{
+					Name:            "wait-for-sidecar",
+					Image:           "busybox",
+					ImagePullPolicy: v1.PullIfNotPresent,
+					Args:            []string{"-c", "until nc -vz $HOST_IP 6060; do echo \"Waiting for local sidecar to listen to $HOST_IP:6060\"; sleep 2; done;"},
+					Command:         []string{"sh"},
+					Env:             env,
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:             sharedVolumeName,
+							MountPath:        "/outputs",
+							MountPropagation: &mountPropagationMode,
+						},
+					},
+					Resources: v1.ResourceRequirements{
+						Limits: v1.ResourceList{
+							v1.ResourceMemory: resource.MustParse("10Mi"),
+							v1.ResourceCPU:    resource.MustParse("10m"),
+						},
+					},
+				},
```

Comment:
Ideally this should be behind a flag if the `testplan` needs `sidecar` or not.

Reply:
If we don't need the sidecar, maybe it will be necessary to add a `wait-for-flannel` initContainer.

Reply:
On a second thought, this is just fine - every host on Kubernetes must have a `sidecar`, so it is fine to wait for it.